src/libraries/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/NativeMemoryTests.cs

```diff
@@ -0,0 +1,162 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Xunit;
+
+namespace System.Runtime.InteropServices.Tests
+{
+    public unsafe class NativeMemoryTests
+    {
+        [Fact]
+        public void AlignedAllocTest()
+        {
+            void* ptr = NativeMemory.AlignedAlloc(1, 1);
+            Assert.True(ptr != null);
+            NativeMemory.AlignedFree(ptr);
+        }
+
+        [Fact]
+        public void AlignedAllocOOMTest()
+        {
+            Assert.Throws<OutOfMemoryException>(() => NativeMemory.AlignedAlloc(1, nuint.MaxValue));
+        }
+
+        [Fact]
+        public void AlignedAllocZeroAlignmentTest()
+        {
+            Assert.Throws<OutOfMemoryException>(() => NativeMemory.AlignedAlloc(0, 0));
+        }
+
```

Comment:
Do we also need tests for the non-power-of-2 alignment, size not multiple of alignment, etc.?

Reply:
Yes and I've added a couple simple tests covering this.